go-controller/pkg/clustermanager/userdefinednetwork/controller.go

```diff
@@ -66,6 +66,9 @@ type Controller struct {
 	namespaceTrackerLock sync.RWMutex
 	// renderNadFn render NAD manifest from given object, enable replacing in tests.
 	renderNadFn RenderNetAttachDefManifest
+	// createPrimaryNetworkLock lock should held when primary network should be created
+	// to avoid having two components trying to create primary network in the same namespace.
```

Comment:
what 2 components are you worried about here? Are you worried someone creates a cUDN and a namespace UDN at same time and in our code we are trying to create both?

Reply:
Couple of questions/comments:
1. Why would we limit this protection to just primary network? Secondary networks could have the same problem.
2. By using this mutex we are serializing  NAD creation across all namespaces. I guess for now this is OK, but in the future if we want to improve perf we may want to use a different mechanism.